roborazzi-accessibility-check/src/main/java/com/github/takahirom/roborazzi/RoborazziATFAccessibilityChecker.kt

```diff
@@ -273,3 +275,26 @@ data class AccessibilityCheckAfterTestStrategy(
       )
   }
 }
+
+@ExperimentalRoborazziApi
+data class AccessibilityCheckEachScreenshotStrategy(
```

Comment:
Thanks. I have a question about the current implementation. 
I haven't read the code thoroughly. The AccessibilityCheckEachScreenshotStrategy is only for RoborazziRule.Options.captureRule and not for captureRoboImage(), right? 
I believe that if we provide the RoborazziContext, we can check every captureRoboImage. Do you have any thoughts on this?

Reply:
I'm happy to take your advice, I'm not as familiar with all the different APIs.

Reply:
Really the intention is to run when roborazzi produces distinct screenshots.

From
- once per test
- per screenshot
- none (roborazzi or a11y checks not enabled)

What gets us closer to that?

Reply:
I think we can add `ruleAccessibilityCheckStrategy` and use it in `captureRoboImage()` like `provideRoborazziContext().accessibilityCheckStrategy.afterScreenshot()` to check the accessibility of each screenshot now that you added `onEach{}`.

https://github.com/takahirom/roborazzi/pull/557/commits/77fa9dc1dc9e68a797cc942405a9134dd52a27df#diff-84240ff001c21726c87098748f05f02f55cb70b8425874729662bc07ad8786ecR66

Reply:
Should I move that type from rule to core module?

Reply:
I've checked, and we can't use `CaptureRoot` because it is in a rule module. 👀

It's a little confusing, so I've renamed it to `RuleCaptureRoot`.

I think we can use `RoboComponent`, but it currently doesn't have fields for `View` or `SemanticsNode`.

However, I realize that this lazy block already retains the instance, and I'm using `RoboComponent.Screen` extensively. I believe we can add the fields to `RoboComponent.View` and `RoboComponent.Compose`. The `RoboComponent` should be collected by the garbage collector, so it should be fine.

So I think we can do it like this. But there might still be issues we have to address.

* Add `View` and `SemanticsNode` fields to `RoboComponent`

* Add `runAccessibilityChecks()` function to use `RoboComponent` (The `runAccessibilityChecks` might be needed for an after-test rule.)

* Change `afterScreenshot`'s parameter from `CaptureRoot` to `RuleCaptureRoot`

Reply:
This feels like a big change I won't get to until next weekend.

Options

- I'll take a look and make changes then
- feel free to make the changes yourself before then
- I can remove that screenshot mode, and just land the other changes for now

No rush, just letting you know where I am

Reply:
I started to make the change and noticed some are already there. I'm unclear if you made all the changes already?

But for RoboComponent, what to do about Screen? Since it has multiple roots?

Reply:
I'm not sure, but can we conduct checks for each root? I don't have a good idea for that, but it might work.

Reply:
Or just the top window?

Reply:
Sounds good. We could output logs like:
"Roborazzi: It seems we have multiple windows, so we'll check accessibility for the top window only."